BDArmory/FX/ExplosionFX.cs

```diff
@@ -524,8 +547,51 @@ private bool IsInLineOfSight(Part part, Part explosivePart, out RaycastHit hit,
                     if (intermediateParts)
                     {
                         var partHP = partHit.Damage();
-                        if (ProjectileUtils.IsArmorPart(partHit)) partHP = 100;
+                        if (ProjectileUtils.IsArmorPart(partHit)) partHP = BDArmorySettings.EXP_PEN_RESIST_MULT * 100;
                         var partArmour = partHit.GetArmorThickness();
+                        var Armor = part.FindModuleImplementing<HitpointTracker>();
+                        if (Armor != null)
+                        {
+                            if (warheadType == WarheadTypes.ShapedCharge)
+                            {
+                                partArmour *= Armor.HEATEquiv;
+                            }
+                            else
+                            {
+                                partArmour *= Armor.HEEquiv;
+                            }
+
+                            float factor = 1.0f;
+
+                            // Exponentiation by Squaring
```

Comment:
This is calculating `Mathf.Pow(1.05, ii)` repeatedly for non-negative integer values of `ii`. A more efficient implementation that only calculates each power once is
```csharp
static List<float> _exp105 = new List<float>() { 1f };
static float Exp105(int power)
{
    if (power >= _exp105.Count)
    {
        for (int p = _exp105.Count; p < power + 1; ++p)
            _exp105.Append(Mathf.Pow(1.05f, p));
    }
    return _exp105[power];
}
```

Reply:
I just replaced this whole bit with taking the variable factor out of the loop and multiplied factor by 1.05f after it was used